server/src/test/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitorTests.java

```diff
@@ -281,30 +324,41 @@ private boolean nodeExceedsQueueLatencyThreshold(NodeUsageStatsForThreadPools no
     }
 
     private TestState createRandomTestStateThatWillTriggerReroute() {
+        int numberOfNodes = randomIntBetween(3, 10);
+        int numberOfHotSpottingNodes = numberOfNodes - 2; // Leave at least 1 non-hot-spotting node.
+        return createTestStateWithNumberOfNodesAndHotSpots(numberOfNodes, randomIntBetween(0, 5), numberOfHotSpottingNodes);
+    }
+
+    private TestState createTestStateWithNumberOfNodesAndHotSpots(
+        int numberOfIndexNodes,
+        int numberOfSearchNodes,
+        int numberOfHotSpottingNodes
+    ) {
         final long queueLatencyThresholdMillis = randomLongBetween(1000, 5000);
```

Comment:
Nit: should we assert that numberOfHotSpottingNodes <= numberOfIndexNodes

Reply:
Done 👍